src/GraphODataTemplateWriter/Extensions/OdcmModelExtensions.cs

```diff
@@ -327,6 +327,70 @@ public static bool IsBaseAbstractAndReferencedAsPropertyType(this OdcmClass comp
             }
         }
 
+        /// <summary>
+        /// Indicates whether class has any properties that are enums
+        /// </summary>
+        /// <param name="complex">The ComplexType that we want to query whether its base type is the referenced type for any property in any entity.</param>
+        /// <returns></returns>
+        public static bool HasEnumProperties(this OdcmClass complex)
+        {
+            return complex.Properties.Any(property => property.Type.AsOdcmEnum() != null);
+        }
+
+        /// <summary>
+        /// Indicates whether the entity's base type is referenced as the type of another property.
+        /// TODO: remove filters on "entity" once we have annotation support.
+        /// </summary>
+        /// <param name="odcmClass">The OdcmClass to inspect.</param>
+        /// <returns>A value of true indicates that OdcmClass's base type is referenced as the type
+        /// of another property in a different entity or complex type.</returns>
+        public static bool IsBaseReferencedAsPropertyType(this OdcmClass odcmClass)
+        {
+            if (odcmClass.Base == null)
+            {
+                return false;
+            }
+            else
+            {
+                var isReferencedInAction = odcmClass.Namespace.Classes
+                    .SelectMany(c => c.Methods)  
+                    .Where( m => !m.IsFunction) // only get the Actions
+                    .Any(m => m.Parameters
+                        .Any( param => param.Type.Name == odcmClass.Base.Name 
```

Comment:
could casing differences derail this first comparison?

Reply:
Agreed. Will action and resolve.

Reply:
Resolved via https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/562